source/ide/wiki/wiki_methods.bas

```diff
@@ -1068,6 +1075,51 @@ FUNCTION wikiSafeName$ (page$) 'create a unique name for both case sensitive & i
     wikiSafeName$ = page$ + "_" + ext$
 END FUNCTION
 
+$UNSTABLE:HTTP
+FUNCTION wikiDLPage$ (url$, timeout!)
+'--- set default result & avoid side effects ---
+wikiDLPage$ = ""
+wik$ = url$: tio! = timeout!
+'--- open client ---
+retry:
+ch& = _OPENCLIENT(wik$)
+IF Help_Recaching < 2 THEN 'avoid messages for 'qb64pe -u' (build time update)
+    IF ch& = 0 AND LCASE$(LEFT$(wik$, 8)) = "https://" THEN
+        IF _MESSAGEBOX("QB64-PE Help", "Can't make secure connection (https:) to Wiki, shall the IDE use unsecure (http:) instead?", "yesno", "warning" ) = 1 THEN
+            IF _MESSAGEBOX("QB64-PE Help", "Do you wanna save your choice permanently for the future?", "yesno", "question" ) = 1 THEN
+                wikiBaseAddress$ = "http://" + MID$(wikiBaseAddress$, 9)
+                WriteConfigSetting generalSettingsSection$, "WikiBaseAddress", wikiBaseAddress$
+            END IF
+            wik$ = "http://" + MID$(wik$, 9): GOTO retry
+        END IF
+    END IF
+END IF
+IF ch& = 0 THEN EXIT FUNCTION

```

Comment:
You can check `_StatusCode` immediately after you know `ch& <> 0`, so you could do it here rather than in the loop, it might simplify things a bit.